erpnext/public/js/education/lms/quiz.js

```diff
@@ -43,6 +57,38 @@ class Quiz {
 		else {
 			this.make_actions();
 		}
+		window.addEventListener('beforeunload', (event) => {
+			event.preventDefault();
+			event.returnValue = '';
+		});
+	}
+
+	set_timer(duration) {
+		this.time_left = duration;
+		var self = this;
+		this.calculate_and_display_time(this.time_left, self, "Time Left - ");
+		this.timer = setInterval(function () {
+			self.time_left -= 1;
+			self.time_taken += 1;
+			self.calculate_and_display_time(self.time_left, self, "Time Left - ");
```

Comment:
I do not think we need to pass the context(`self`) because you are calling the class method on the object